consensus/polybft/fsm.go

```diff
@@ -172,6 +176,29 @@ func (f *fsm) BuildProposal(currentRound uint64) ([]byte, error) {
 		}
 
 		extra.Validators = f.newValidatorsDelta
+
+		if f.config.IsBridgeEnabled() {
+			signature, err := bls.UnmarshalSignature(extra.Committed.AggregatedSignature)
```

Comment:
Move all of this signature preparation to the `createValidatorSetCommit` function and only call it if the `newValidatorsDelta` is not empty.